### ISSUE TITLE
Update physics API and example

### ISSUE THREAD

**Issue by Aposhian** [joshuajbouw/bevy_tilemap#176]
**What problem does this solve or what need does it fill?**
Rapier physics integration is very valuable, but the current API and examples do not use the latest API, and `physics_dungeon` is broken.

**Describe the solution would you like?**
Use the latest Rapier API including:
- Using RigidBody and Collider components rather than `RigidBodySet`

**Describe the alternative(s) you've considered?**
Removing the physics example while non-functional?

**Additional context**
N/A

**Comment by joshuajbouw** [comment]
Yeah it was shelved for the time being as it required quite a large overhaul that was WIP.
